### DISCUSSION TITLE
Need help compiling companion

### DISCUSSION THREAD

**Discussion by JustinKScott** [EdgeTX/edgetx#4115] [General]
Good evening!

I'dl like to compile companion so I can debug my lua mixers.   After excessive Googling; this is the closest I've figured out... but it's not working.  (Note: I'm using the newly fixed GitPod link)

cmake -Wno-dev -DPCB=XLITES -DLUA_MIXER=ON -DHELI=OFF -DFLIGHT_MODES=OFF -DCMAKE_BUILD_TYPE=Release ../
make configure
make -C native -j$(nproc) companion
// Then go collect my build.

However the final make step yields this, and I'm not sure where to go next:

![Screenshot 2023-09-26 at 11 39 18 PM](https://github.com/EdgeTX/edgetx/assets/5257743/c3ad63ea-d841-4116-b662-46e98994399d)

**Comment by pfeerick** [comment]
The answer is unfortunately simple  - you cannot build companion on using GitPod using that docker container - it is for firmware building only. Even if it were possible, it would 1) be a Linux build of Companion that may or may not run on your own system if you run Linux (due to library dependencies, etc) and 2) take forever to build. 

I take it that what you actually want is a simulator build with LUA_MIXER enabled? As I'm not sure compiling Companion with `LUA_MIXER=ON` is going to do anything different to Companion - it should be compiled with all supported options turned on.

**Reply by JustinKScott** [reply]
Hmmm.. thank you.

So, I have companion 2.8 release on my mac, but unfortunately it's simulator doesn't show the mixer screens.  Is there a way to enable them?  Perhaps there now a new way to debug LUA mixers in edgeTX?  In optx; companion was the way I used to find my mixer errors.

All I really need is the debug output window; can I get that some how directly from my firmware device?

**Reply by pfeerick** [reply]
Unfortunately, it seems not - the X-Lite doesn't seem to have the ability to do a virtual COM port else you could possibly have been able to debug over USB. 

Simulator follows the firmware options, and on 2.9, the LUA_MIXER is only enabled by default for colorlcd radios (as generally the B&W radios don't have enough flash memory to have that option turned on across the board, and I wasn't going to get into mess that is conditional compilation on the main firmware line). 

Since you're on Mac, maybe @raphaelcoeffic can help you out with a Mac-compatible simulator library build for the XLite using the above compile flags unless he has a better idea ;)

**Comment by WillyZehnder** [comment]
I have the same problem as @JustinKScott after building the firmware for my QX7 and X9E with -DLUA_MIXER=Y using Gitpod.

Some time went by and my question is, if there are any new possibilities? Is it possible to build Companion for Windows in Gitpod in meantime?

Maybe the Companion is in general built with all possible options for all transmitters, 'cause there are no memory restrictions?

**Reply by raphaelcoeffic** [reply]
> Is it possible to build Companion for Windows in Gitpod in meantime?

No, and it is not planned to be supported.

Is this about having simulator plugins for specific radios having specific options? If yes, there is an on-going effort to make simulator plug-ins cross-platform and compilable from Linux for all platforms (including executing in web browser).

**Comment by WillyZehnder** [comment]
> Is this about having simulator plugins for specific radios having specific options? If yes, there is an on-going effort to make simulator plug-ins cross-platform and compilable from Linux for all platforms (including executing in web browser).

I think that's the better way - Thank you very much for you great work!

Is there a place or a link where I can see the development and eventually early test that plug-ins?
